README.md

```diff
@@ -1,51 +1,59 @@
-        4. [Validation](#validation)
-        5. [Documentation](#documentation)
-        6. [Swagger](#swagger)
-    2. [Build image](#build-image)
-    3. [Server Wrapper](#server-wrapper)
-    4. [Database Migrations](#db-migrations)
-    5. [Example](#example)
-2. [REST API Syntax Specification](#rest-api-syntax-specification)
-    1. [Resources and Collections](#resources-and-collections)
-    2. [HTTP Headers](#http-headers)
-    3. [Responses](#responses)
-    4. [Errors](#errors)
-    5. [Collection Operators](#collection-operators)
-        1. [Field Selection](#field-selection)
-        2. [Sorting](#sorting)
-        3. [Filtering](#filtering)
-        4. [Pagination](#pagination)
-    6. [Field Presence](#field-presence)
-
-## Getting Started
-
-Toolkit provides a means to have a generated code that supports a certain common functionality that
-is typically required for any service.
-Toolkit declares its own format for Responses, Errors, Long Running operations, Collection operators.
-More details on this can be found in appropriate section on this page.
-
-Toolkit approach provides following features:
-- Application may be composed from one or more independent services (micro-service architecture)
-- Service is supposed to be a gRPC service
-- REST API is presented by a separate service (gRPC Gateway) that serves as a reverse-proxy and
-forwards incoming HTTP requests to gRPC services
-
-### Initializing your Application
+This repository provides common Go utilities and helpers that are reusable from project-to-project. The goal is to prevent code duplication by encouraging teams to use and contribute to toolkit libraries.
+
+The toolkit is not a framework. Rather, it is a set of (mostly gRPC-related) plugins and helpers.
+
+## Background
+
+The toolkit's approach is based on the following assumptions.
+
+- An application is composed of one or more independent services (micro-service architecture)
+- Each independent service uses gRPC
+- The REST API is presented by a separate service (gRPC Gateway) that serves as a reverse-proxy and forwards incoming HTTP requests to gRPC services
 
 To get started with the toolkit, check out the [Atlas CLI](https://github.com/infobloxopen/atlas-cli) repository. The Atlas CLI's "bootstrap command" can generate new applications that make use of the toolkit. For Atlas newcomers, this is a great way to get up-to-speed with toolkit best practices.
 
-### Plugins
+## Toolkit Libraries
+
+The following libraries have how-to guides included at the package level.
+
+#### Request Handling Tools
+
+[`requestid`](requestid) - gets the request ID from incoming requests (or creates a unique ID if it doesn't exist) 
+
+[`query`](query) - provides query parameter-specific helpers, like sorting, paging, and filtering resources
+
+[`auth`](auth) - includes authorization and authentication-related helpers
+
+[`rpc`](rpc) - modifies gRPC responses before they are returned to users
```

Comment:
Package has some logic for error handling and resource identifiers at the moment. A better description is needed but we can think of it once reviewing `rpc` package readme.

Reply:
> A better description is needed but we can think of it once reviewing rpc package readme.

Sounds good to me!